Remove security parameter (#93)
cfrg/draft-irtf-cfrg-voprf: draft-irtf-cfrg-voprf.md

- `H_1`: A hash function mapping two fixed-length byte arrays to another
  byte array of fixed-length depending on security requirements.

If a ciphersuite corresponds to an instantiation of the protocol in the
verifiable setting (VOPRF), then it will contain instantiations of the
following functions.

- `H_2`: Maps an arbitrary-length sequence of bytes to a Scalar value in
  `GF(p)`, where `p = GG.Order()`.
- `H_3`: Maps an arbitrary-length sequence of bytes to a another byte
  array of fixed-length depending on security requirements.

Specific instantiations of these ciphersuites are given in
{{ciphersuites}}.

## Setup {#setup}

Before the protocol takes place, the Server MUST select a valid
ciphersuite from the list in {{ciphersuites}}. Once a selection is made,
it publishes the ciphersuite that it is using to make it available to
any Client that connects to it.

The Server SHOULD sample a new PrivateKey object corresponding to its
secret input. This value can be used across multiple instantiations of
the protocol. If the server ciphersuite supports verifiability, then it
interprets its private key as a Scalar `k` in `GF(p)` and computes:

~~~
PK = k * GG.Generator()
~~~

> Servers that support verifiability MUST make public_key available to

Reviewer: ```suggestion
Servers that support verifiability MUST make `PK` available to
```

- `H_1`: A hash function mapping two fixed-length byte arrays to another
  byte array of fixed-length depending on security requirements.

If a ciphersuite corresponds to an instantiation of the protocol in the
verifiable setting (VOPRF), then it will contain instantiations of the
following functions.

- `H_2`: Maps an arbitrary-length sequence of bytes to a Scalar value in
  `GF(p)`, where `p = GG.Order()`.
- `H_3`: Maps an arbitrary-length sequence of bytes to a another byte
  array of fixed-length depending on security requirements.

Specific instantiations of these ciphersuites are given in
{{ciphersuites}}.

## Setup {#setup}

Before the protocol takes place, the Server MUST select a valid
ciphersuite from the list in {{ciphersuites}}. Once a selection is made,
it publishes the ciphersuite that it is using to make it available to
any Client that connects to it.

The Server SHOULD sample a new PrivateKey object corresponding to its
secret input. This value can be used across multiple instantiations of
the protocol. If the server ciphersuite supports verifiability, then it
interprets its private key as a Scalar `k` in `GF(p)` and computes:

~~~
PK = k * GG.Generator()

> public_key = GG.Serialize(PK)

Reviewer: Why `public_key` for the serialized version and `PK` for the deserialized one?

~~~

The Server MUST publish `public_key` so that any valid Client can use it

> during the protocol to verify messages from Server.

Reviewer: This seems like a weird MUST. What if there's an application that genuinely does not require verifiability? I think we should reframe this a bit, e.g., "Servers that support verifiability MUST make `public_key` available to clients." (And I'd drop the serialization requirement at the same time.)

Steps:

 1. DST = "oprf_derive_output"
 2. dk = H_1(DST, T.data || E)
 3. output = H_1(dk, aux)
 4. Output output
~~~

## Fixed-base blinding {#blinding}

> Let `H` refer to the function `GG.HashToGroup`, in {{pog}} we assume

Reviewer: This section seems to revert to the old scalar multiplication notation. Is that an intentional regression?

Author: Will fix

 4. Output unblinded_tokens
~~~

Let `P = GG.HashToGroup(x)`. Notice that Unblind computes:

~~~

> (Z-PKR) = k(P + r * G) - (rk) * G = k * P

Reviewer: ```suggestion
Z - PKR = k(P + r * G) - (rk) * G = k * P
```

## Cryptographic security {#cryptanalysis}

We discuss the cryptographic security of the (V)OPRF protocol from
{{protocol}}, relative to the necessary cryptographic assumptions that
need to be made.

### Computational hardness assumptions {#assumptions}

Each assumption states that the problems specified below are

> computationally difficult to solve in relation to a paritcular choice of

Reviewer: ```suggestion
computationally difficult to solve in relation to a particular choice of
```

> security parameter `sp`. In other words, the probability that an
> adversary has in solving the problem is bounded by a function negl(sp),
> where negl(sp) < 1/f(sp) for all polynomial functions f().

Reviewer: ```suggestion
security parameter `sp`.
```
